src/lib.rs

```diff
@@ -3,7 +3,9 @@ use std::os::raw::c_uchar;
 
 use crate::go::slice::{GoSlice, ToGoSlice};
 use crate::go::string::{GoString, ToGoString};
-use crate::go::guess::{GoGuess, Guess};
+use crate::go::guess::GoGuess;
+
+pub use go::guess::Guess;
```

Comment:
I'm no Rust expert and am curious why re-exporting Guess is required in there, i.e. what does it mean  
> interfaces that rely on this type are totally unusable
?

Reply:
Your doubt is right, my statement is not technically correct, sorry for being inaccurate here. 

The `Guess` type is indeed exposed and thus public, but its parent module `src/go/guess.rs` is **not** (`pub(crate) mod guess`, it is only visible within the crate) so those interfaces relying on it are usable, but users are not allowed to explicitly access it.

Since the `Guess` type is public, there are no compile errors or warnings.

------

However, it is still recommended to re-export the `Guess` type as it is used by public interfaces

Reply:
Thank you for the explanation. I'm a bit reluctant to extend public API surface without clear understand what specifically does the lack of re-export preclude the clients from.

But thinking more about it - although we don't have it in Go impl, we do seem to have Guess as part of API at e.g. [Java bindings](https://github.com/go-enry/go-enry/blob/c3899f12caafff51183b1828344b1edbe470981c/java/src/main/java/tech/sourced/enry/Enry.java#L8) so it probably should be fine.